samples/ImplicitFlow/AuthorizationServer/Startup.cs

```diff
@@ -0,0 +1,142 @@
+using AuthorizationServer.Models;
+using AuthorizationServer.Services;
+using CryptoHelper;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OpenIddict;
+using System;
+using System.Linq;
+
+namespace AuthorizationServer
+{
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var configuration = new ConfigurationBuilder()
+                        .AddJsonFile("config.json")
+                        .AddEnvironmentVariables()
+                        .Build();
+
+            services.AddDbContext<ApplicationDbContext>(options =>
+            {
+                options.UseSqlite("Filename=./authorization.db");
+            });
+
+            services.AddIdentity<ApplicationUser, IdentityRole<Guid>>()
+                .AddEntityFrameworkStores<ApplicationDbContext, Guid>()
+                .AddDefaultTokenProviders();
+
+            // Register the OpenIddict services, including the default Entity Framework stores.
+            services.AddOpenIddict<ApplicationUser, IdentityRole<Guid>, ApplicationDbContext, Guid>()
+
+                // Enable the authorization, logout, userinfo, and introspection endpoints.
+                .EnableAuthorizationEndpoint("/connect/authorize")
+                .EnableLogoutEndpoint("/connect/logout")
+                .EnableUserinfoEndpoint("/connect/userinfo")
+                .EnableIntrospectionEndpoint("/connect/introspect")
+
+                // Note: the Mvc.Client sample only uses the authorization code flow but you can enable
+                // the other flows if you need to support implicit, password or client credentials.
+                .AllowImplicitFlow()
+
+                // During development, you can disable the HTTPS requirement.
+                .DisableHttpsRequirement()
+
+                // Register a new ephemeral key, that is discarded when the application
+                // shuts down. Tokens signed using this key are automatically invalidated.
+                // This method should only be used during development.
+                .AddEphemeralSigningKey();
+
+            services.AddCors();
+            services.AddMvc();
+
+            services.AddTransient<IEmailSender, AuthMessageSender>();
+            services.AddTransient<ISmsSender, AuthMessageSender>();
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILogger<Startup> logger)
+        {
+            app.UseIdentity();
+            app.UseOAuthValidation();
+
+            app.UseCors(builder => builder
```

Comment:
Hmm. I don't need it for the discovery system to work, because I can register just the client that I want to be able to do discovery. I guess that in the big picture, we want any application to be able to do discovery. For this example, though, I can restrict discovery to just the client application. It's good to know that CORS only happens with browser requests.